File: C1.UWP.ExpressionEditor/VB/ExpressionEditorSamples/FlexGridEE.vb
Imports C1.Xaml.FlexGrid
Imports C1.Xaml.ExpressionEditor

Public Class FlexGridEE
    Inherits C1FlexGrid
    Implements ISupportExpressions
#Region "Ctor"
    Public Sub New()
        MyBase.New()
        ExpressionEditors = New ExpressionEditorCollection(Me)
    End Sub
#End Region

#Region "ISupportExpressions"

    Public Sub SetCellValue(row As Integer, colName As String, value As Object) Implements ISupportExpressions.SetCellValue
        Me(row, colName) = value
    End Sub

    Public Overloads ReadOnly Property ExpressionEditors() As ExpressionEditorCollection Implements ISupportExpressions.ExpressionEditors

    Private Property ISupportExpressions_ItemsSource As IEnumerable Implements ISupportExpressions.ItemsSource
        Get
            Return Me.ItemsSource
        End Get
        Set(value As IEnumerable)
            Me.ItemsSource = value
        End Set
    End Property
#End Region
End Class

File: C1.UWP.ExpressionEditor/CS/ExpressionEditorSamples/FlexGridEE.cs
using C1.Xaml.ExpressionEditor;
using C1.Xaml.FlexGrid;

namespace ExpressionEditorSamples
{
    public class FlexGridEE : C1FlexGrid, ISupportExpressions
    {
        #region Ctor
        public FlexGridEE() : base()
        {
            ExpressionEditors = new ExpressionEditorCollection(this);
        }
        #endregion

        #region Override
        protected override void OnCellEditEnded(CellEditEventArgs e)
        {
            base.OnCellEditEnded(e);
            // ignore expression column changing
            if (!e.Cancel && !ExpressionEditors.Contains(Columns[e.Column].ColumnName))
                ExpressionEditors.Evaluate(GetDataIndex(e.Row));
        }
        #endregion

        #region ISupportExpressions
        public void SetCellValue(int row, string colName, object value)
        {
            this[GetRowIndex(row), colName] = value;
        }

        public ExpressionEditorCollection ExpressionEditors { get; }
        #endregion
    }
}
